src/server/dfly_main.cc

```diff
@@ -206,7 +206,7 @@ void VersionMonitor::RunTask(SSL_CTX* ssl_ctx) {
         GetRemoteVersion(my_pb, ssl_ctx, host_name, port, resource, version_header);
     if (remote_version) {
       const std::string rv = remote_version.value();
-      if (rv != current_version) {
```

Comment:
I'm not sure if this is the best solution. We're using `X.Y.Z` format, and it isn't string sortable. Think of the case of current == `1.6.1` vs rv == `1.12.1`.
I guess we'll need to `absl::StrSplit()` it into 3 parts and compare each?

Reply:
Really good catch. I am not sure how I missed this but I am very happy that you spotted this.